internal/pkg/term/syncbuffer/syncbuffer.go

```diff
@@ -0,0 +1,96 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package syncbuffer provides a goroutine safe bytes.Buffer as well printing functionality to the terminal.
+package syncbuffer
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+)
+
+// syncBuffer is a synchronized buffer that can be used to store output data and coordinate between multiple goroutines.
+type syncBuffer struct {
```

Comment:
i'm realizing this package will be used like 
```
buf := syncbuffer.New()
labeledBuf = buf.WithLabel("asdf")
```

maybe we should only export a `New(label string) *LabeledSyncBuffer` function? it's kinda weird have an exported function return an unexported struct. alternatively, we could just re-export `SyncBuffer` and keep the usage the same (sorry for my prev comment!) - i'll leave it up to you!

Reply:
I reverted back to public